src/server/service/agent.ts

```diff
@@ -59,24 +58,27 @@ export class Agent {
   async triggerResponse() {
     // TODO: perform actual logic to understand if response should be triggered or not
     const shouldTrigger = true;
-
     if (!shouldTrigger) return;
 
     this._triggeredAt = Date.now();
 
-    const message = await this.requestMessageFromLLM();
+    const response = await this.requestMessageFromLLM();
 
     // If inference failed or bot decided not to reply, let the agent be silent
-    if (!message || message.includes(this._silenceToken)) return;
+    if (!response || response.includes(this._silenceToken)) return;
+
+    const cleanResponse = this.parseResponse(response);
 
     const payload: ChatMessagePayload = {
       sender: this.id,
-      message,
+      message: cleanResponse,
       sentAt: Date.now(),
     };
 
     // TODO: remove artificial wait in favour of something more inteligent
-    await wait(1500);
+    if (this._match.messages.length === 1)
```

Comment:
lovely opportunity for a ternary
```ts
await this._match.messages.length === 1
  ? wait(4500)
  : wait(1500)
```
or 
```ts
await wait(this._match.messages.length === 1 ? 4500 : 1500)
```

Reply:
Good one... 
This was lazy last minute fix hehe.